Allow as many CN slots as specified in A1111 Settings page

andyxr opened the pull request:
If the user sets the max CN unit count to something above 5, then Deforum will respect that and have the same number. If they set it below 5, then we keep the minimum Deforum CN slots at 5, in order to prevent any breakages when importing settings files.
This would close the feature request: https://github.com/deforum-art/sd-webui-deforum/issues/812

andyxr committed:
WIP: Support CN max models slider value

WIP
Deforum will always have a min of 5 CN slots, and a max of whatever value is set in A1111's Settings->ControlNet->Max models slider

andyxr committed:
Merge remote-tracking branch 'upstream/automatic1111-webui' into automatic1111-webui

andyxr committed:
No of CNs in Deforum respect max CNs specified in A1111 settings

If the user sets max CNs to be less than 5, then I keep 5 as the minimum in Deforum. If they go above 5, then you get that many CNs in Deforum

andyxr committed:
Fixed a defect in multi-CN support

Found another location where number of CNs was hardcoded as 5. Now addressed.

rewbs commented on scripts/deforum_helpers/animation_key_frames.py:
```diff
@@ -75,7 +76,11 @@ class ControlNetKeys():
         self.fi = FrameInterpolater(max_frames=anim_args.max_frames)
         self.schedules = {}
-        for i in range(1, 6): # 5 CN models in total
+        max_models = shared.opts.data.get("control_net_unit_count", 1) if shared.opts.data.get("control_net_unit_count", 1) is not None else shared.opts.data.get("control_net_max_models_num", 1)
```
I think `shared.opts.data.get("control_net_unit_count", 1) is not None` will evaluate to `true` if `control_net_unit_count` is not set because of the default arg of 1. This means we'll never fall back to looking at `control_net_max_models_num` unless `control_net_unit_count` is explicitly set to `None` in `opts.data`. Is that the intent?

andyxr replied:
Ah OK, will rectify. The idea is that if you're on pre-1.6 A1111 ,that control_net_unit_count won't be there. I made a mistake in setting the default arg to 1. Good catch!

rewbs reviewed (approved):
Looks good to me. Have played with and it seems to work great. My only question is about the fallback to `control_net_max_models_num` as per inline comment.

andyxr committed:
Addressed issue where default value for max_models was wrong

As per Rewbs' suggestion, I fixed the code so that it's a little more inline and the default is acceptable
